pythran/pythonic/types/complex.hpp

```diff
@@ -63,4 +63,38 @@ namespace pythonic
   }
 }
 
+#ifdef ENABLE_PYTHON_MODULE
+
+#include "pythonic/python/core.hpp"
+
+namespace pythonic
+{
+
+  template <class T>
+  struct to_python<std::complex<T>> {
+    static PyObject *convert(std::complex<T> const &c)
+    {
+      return PyComplex_FromDoubles(c.real(), c.imag());
+    }
+  };
+
+  template <class T>
+  struct from_python<std::complex<T>> {
+    static bool is_convertible(PyObject *obj)
+    {
+      return PyComplex_Check(obj) or PyFloat_Check(obj) or PyInt_Check(obj);
+    }
+    static std::complex<T> convert(PyObject *obj)
+    {
+      if (PyComplex_Check(obj))
+        return {PyComplex_RealAsDouble(obj), PyComplex_ImagAsDouble(obj)};
+      else if (PyFloat_Check(obj))
+        return {PyFloat_AsDouble(obj), 0.};
+      else
+        return {(double)PyInt_AsLong(obj), 0.};
```

Comment:
Here you check type twice. Once in `is_convertible` and another in `convert`. Why not using only `convert` and return nullptr when it fail?